File: БР1.1/Захматов Юрий/labs/lab1/src/main/java/com/petproject/itmoacsbackend/users/controllers/UserController.java
package com.petproject.itmoacsbackend.users.controllers;

import com.petproject.itmoacsbackend.booking.dto.BookingResponse;
import com.petproject.itmoacsbackend.booking.entities.BookingEntity;
import com.petproject.itmoacsbackend.booking.repositories.BookingRepository;
import com.petproject.itmoacsbackend.booking.service.BookingService;
import com.petproject.itmoacsbackend.property.dto.PropertyResponse;
import com.petproject.itmoacsbackend.property.repositories.PropertyRepository;
import com.petproject.itmoacsbackend.property.service.PropertyService;
import com.petproject.itmoacsbackend.users.dto.UserResponse;
import com.petproject.itmoacsbackend.users.dto.UserUpdateRequest;
import com.petproject.itmoacsbackend.users.entities.UserEntity;
import com.petproject.itmoacsbackend.users.repositories.UserRepository;
import com.petproject.itmoacsbackend.users.service.UserService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.http.ResponseEntity;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.*;

@RestController
@RequiredArgsConstructor
@RequestMapping("/api/v1/users/me")
public class UserController {

    private final UserService userService;
    private final PropertyService propertyService;
    private final BookingService bookingService;


    @GetMapping
    public ResponseEntity<UserResponse> getCurrentUser(
            @AuthenticationPrincipal UserEntity user
    ) {
        UserResponse response = userService.getCurrentUser(user);
        return ResponseEntity.ok(response);
    }

    @PatchMapping
    public ResponseEntity<UserResponse> updateCurrentUser(
            @AuthenticationPrincipal UserEntity user,
            @Valid @RequestBody UserUpdateRequest request
    ) {
        UserResponse response = userService.updateUser(request, user);
        return ResponseEntity.ok(response);
    }

    @GetMapping("/bookings")
    public ResponseEntity<Page<BookingResponse>> getBookings(
            @RequestParam(defaultValue = "0") int page,
            @RequestParam(defaultValue = "20") int size,
            @AuthenticationPrincipal UserEntity user
    ) {
        return ResponseEntity.ok(bookingService.getAllUserBookings(user,page,size));
    }

    @GetMapping("/properties")
    public ResponseEntity<Page<PropertyResponse>> getProperties(
            @RequestParam(defaultValue = "0") int page,
            @RequestParam(defaultValue = "20") int size,
            @AuthenticationPrincipal UserEntity user
    ) {
        return ResponseEntity.ok(propertyService.getAllUserProperties(user, page, size));
    }

}

File: БР1.2/Цатинян Артём/labs/lab1/src/main/kotlin/ru/itmo/restaurantbooking/user/adapter/rest/UserController.kt
package ru.itmo.restaurantbooking.user.adapter.rest

import io.swagger.v3.oas.annotations.security.SecurityRequirement
import org.springframework.security.core.Authentication
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PatchMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import ru.itmo.restaurantbooking.auth.domain.AuthenticatedUser
import ru.itmo.restaurantbooking.common.adapter.rest.config.OpenApiConfig
import ru.itmo.restaurantbooking.user.adapter.rest.dto.UpdateProfileRequest
import ru.itmo.restaurantbooking.user.adapter.rest.dto.UserProfileResponse
import ru.itmo.restaurantbooking.user.service.UserService

@RestController
@RequestMapping("/api/v1/users")
@SecurityRequirement(name = OpenApiConfig.BEARER_AUTH)
class UserController(
    private val userService: UserService
) {

    @GetMapping("/me")
    fun me(authentication: Authentication): UserProfileResponse = userService.getCurrentUser((authentication.principal as AuthenticatedUser).id)

    @PatchMapping("/me")
    fun updateMe(authentication: Authentication, @RequestBody request: UpdateProfileRequest): UserProfileResponse =
        userService.updateCurrentUser((authentication.principal as AuthenticatedUser).id, request)
}
